### DISCUSSION TITLE
Help with Saving Inference Output

### DISCUSSION THREAD

**Discussion by lenlobo** [Project-MONAI/MONAI#6802] [Q&A] [answered]
Hi,
I have a UNet which predicts with 2 classes, either an anomaly or not. So 0 or 1.
When I run the inference and I do
test_data['pred'].shape, I get (2,300,300,300)
 Is it possible to use AsDiscreted and SaveImaged functions and save only the (1,300,300,300) slice? Because when I observe in MITK, I get another 4th dim slider at the bottom which lets me switch between. But I would like to avoid this.
Thank you

**Comment by KumoLiu** [comment, marked as answer]
Hi @lenlobo, you can use `SplitDimd` after `AsDiscreted` and only save maybe `pred_1` in `SaveImaged`.
https://github.com/Project-MONAI/MONAI/blob/c4ff70b3df52f16e4a2155217f8e5f3aa6be8ce6/monai/transforms/utility/dictionary.py#L382

I hope it helps, thanks!

**Reply by lenlobo** [reply]
@KumoLiu Thank you so much! You've been an amazing help to me :)
